### PR TITLE
method to create an affiliate

### PR DESCRIPTION
**Pull request by shekhardokania**

### CODE REVIEW

File: spec/lib/affiliate_spec.rb [hunk 1]
```diff
@@ -47,4 +47,21 @@
       validate_call response
     end
   end
+
+  describe 'create' do
+    context 'when required params are missing' do
+      it 'should raise exception' do
```
Comment (psychocandy) [inline spec/lib/affiliate_spec.rb]: I would split this into 2 specs, one for missing company one for missing zipcode. 
Would be easier to track if there will be more attributes in the future.

File: spec/lib/affiliate_spec.rb [hunk 2]
```diff
@@ -47,4 +47,21 @@
...
+        expect { subject.create company: "xyz@gmail.com", status: "active" }.to raise_error ArgumentError
+        expect { subject.create zipcode: "10178", status: "active" }.to raise_error ArgumentError
+      end
+    end
+
+    it 'should make a successful request call' do
+      response = subject.create company: "xyz@gmail.com", zipcode: "10178"
```
Comment (psychocandy) [inline spec/lib/affiliate_spec.rb]: Do you or do you not need `status`? if we don't why is it in the previous spec?

**Comment by psychocandy** [comment]
minor comments, otherwise :+1:

File: spec/lib/affiliate_spec.rb (hunk 2 above)
Comment (shekhardokania) [inline spec/lib/affiliate_spec.rb]: No we don't need the status. It's just there as an extra param :)

File: spec/lib/affiliate_spec.rb (hunk 1 above)
Comment (shekhardokania) [inline spec/lib/affiliate_spec.rb]: :+1:

**Pull request merged**